nengo/builder/tests/test_builder.py

```diff
@@ -51,3 +56,54 @@ class Test:
     my_builder(1)
     with pytest.warns(Warning, match="Type .* already has a builder. Overwriting"):
         my_builder(1)  # repeat setup warning
+
+
+@pytest.mark.skipif(sys.version_info < (3, 6, 0), reason="requires ordered dicts")
```

Comment:
If we use `OrderedDict` in `OrderedSet` then we shouldn't need this.

Reply:
Unfortunately not, there's other places that we use regular `dicts` within the build process that introduce non-determinism, besides `sets`. And fixing those didn't seem worth it, since we'll be dropping Python 3.5 support very soon.

Reply:
So I looked into making this deterministic in Python 3.5, and it's actually quite difficult. There are a lot of places where dict order can randomize things, including in e.g. `NeuronType.state`, other parts of making the graph, etc. I haven't been able to track them all down, and I don't think it's worth putting more time into. Looking at the NengoDL PR, things might actually be deterministic enough that the previously broken functionality there is now fixed for Python 3.5 as well, even though things aren't completely deterministic.

So we'll just leave this full determinism as something we don't support in Python 3.5.